numpy/core/numeric.py

```diff
@@ -2279,19 +2279,21 @@ def within_tol(x, y, atol, rtol):
         return cond[()]  # Flatten 0d arrays to scalars
 
 
-def _array_equal_dispatcher(a1, a2):
+def _array_equal_dispatcher(a1, a2, equal_nan=None):
     return (a1, a2)
 
 
 @array_function_dispatch(_array_equal_dispatcher)
-def array_equal(a1, a2):
+def array_equal(a1, a2, equal_nan=False):
     """
     True if two arrays have the same shape and elements, False otherwise.
 
     Parameters
     ----------
     a1, a2 : array_like
         Input arrays.
+    equal_nan : bool
+        Whether to compare NaN's as equal.
 
```

Comment:
Sorry, bad case of not-careful enough first review. This should have a `.. versionadded:: 1.19.0` tag.

Reply:
Yup, forgot that myself. Does it need a release note as well?